### DISCUSSION TITLE
How To Makde sure Environment Variables Is Loaded When Running Test (Vitest)

### DISCUSSION THREAD

**Discussion by DevYemi** [vercel/next.js#62021] [Help]
Hi there,

So i am trying to write some unit tests for some component in my nextJs app but my  environment variables keeps returning `undefined`. The environment variables works as expected in dev mode so this only happens when i run my tests, also i don't want to define an `.env.test` i just want my tests to read the environment variables in my `.env.local` file

I have tried different solutions like one suggested in this issue [here](https://github.com/vercel/next.js/issues/17903#issuecomment-733994841)

````
// In my vitest set-up tests file
import { loadEnvConfig } from '@next/env'

export default async () => {
  loadEnvConfig(process.env.PWD)
}
````

i have also tried:
````
import next from 'next'
next({})
````

But all that didn't work. I have also seen a solution online asking me to use the `dot` package but i would like to know if there isn't a cleaner or better wat to do this ?

**Comment by TheRealOwenRees** [comment]
Hello,

when running tests, `NODE_ENV` is set to `test`. When this happens, `.env.local` is not read. You will need a `.env.test` file for this to work.

Below is my `vitest.config.ts` file, which when coupled with `.env.test` works fine.

```js
import { loadEnvConfig } from "@next/env";
import react from "@vitejs/plugin-react";
import { defineConfig } from "vitest/config";

loadEnvConfig(process.cwd());

export default defineConfig({
  plugins: [react()],
  test: {
    environment: "jsdom",
  },
  resolve: {
    alias: {
      "@": "/src",
    },
  },
});

```

I hope this helps, only 4 months later!
